extension/src/repository/index.ts

```diff
@@ -89,7 +95,9 @@ export class Repository extends DeferredDisposable {
   }
 
   private setState() {
-    this.sourceControlManagement.setState(this.getState())
-    this.decorationProvider.setState(this.getState())
+    this.sourceControlManagement.setState(
+      this.model.getSourceControlManagementState()
+    )
+    this.decorationProvider.setState(this.model.getDecorationState())
```

Comment:
[F] Bug two. 

Was a great example of trying to do something clever that turned out to be incredibly dumb.

Previous code:

```
export type DecorationState = Record<Status, Set<string>>

export interface DecorationModel {
  getState: () => DecorationState
}

enum Status {
  ADDED = 'added',
  DELETED = 'deleted',
  MODIFIED = 'modified',
  NOT_IN_CACHE = 'notInCache',
  RENAMED = 'renamed',
  GIT_MODIFIED = 'gitModified',
  TRACKED = 'tracked'
}

export class DecorationProvider
  extends Disposable
  implements FileDecorationProvider
{
...

  public setState(state: DecorationState) {
    const urisToUpdate = this.getUnion(this.state, state)
    this.state = state
    this.decorationsChanged.fire(urisToUpdate)
  }

  private getUnion(existingState: DecorationState, newState: DecorationState) {
    return flattenUnique([
      ...Object.values(existingState).map(status => [...(status || [])]),
      ...Object.values(newState).map(status => [...(status || [])])
    ]).map(path => Uri.file(path))
  }
```

```
export class RepositoryModel
  extends Disposable
  implements DecorationModel, SourceControlManagementModel
{
...
  public getState() {
    const acc = []
    for (const relTrackedOut of this.relTrackedOuts) {
      acc.push(join(this.dvcRoot, relTrackedOut))
    }

    return {
      ...omit(this.state, ['trackedLeafs', 'trackedNonLeafs']),
      hasRemote: new Set([...acc, ...this.state.trackedLeafs]),
      tracked: this.getTracked()
    }
  }
```

The way that this was setup meant that the compiler couldn't/didn't warn me that I was putting a boolean into `Uri.file` which causes a silently rejected promise and stopped any decorations from being provided.

I've changed the code to be very explicit in terms of what the `DecorationProvider` and `SourceContorlManagement` expect for state.